src/Conversion/ONNXToTOSA/Math/Conv2D.cpp

```diff
@@ -75,81 +73,81 @@ Value createConvInGroups(PatternRewriter &rewriter, Operation *op,
+    auto convOp = llvm::cast<ONNXConvOp>(op);
+
     auto input = adaptor.X();
     auto weights = adaptor.W();
     auto bias = adaptor.B();
 
+    auto inputType = input.getType().cast<TensorType>();
     auto weightType = weights.getType().cast<ShapedType>();
 
-    auto weightShape = weightType.getShape();
+    // Get shapehelper for autopad attributes
+    IndexExprBuilderForTosa createTosaIE(rewriter, convOp->getLoc());
+    NewONNXConvOpShapeHelper shapeHelper(op, operands, &createTosaIE);
+    shapeHelper.computeShapeAndAssertOnFailure();
 
-    StringAttr autopad = adaptor.auto_padAttr();
-    ArrayAttr dilations = adaptor.dilationsAttr();
-    IntegerAttr group = adaptor.groupAttr();
-    ArrayAttr pads = adaptor.padsAttr();
-    ArrayAttr strides = adaptor.stridesAttr();
+    auto weightShape = weightType.getShape();
 
-    Type resultType = getTypeConverter()->convertType(op.getResult().getType());
+    Type resultType = convOp.getResult().getType();
 
-    // NOTE: we would like if inferShapes() had filled in explicit padding
-    // but currently inferShapes() does not do this for ConvOp (it does for
-    // ConvTransposeOp). We have not implemented code for autopad so fail.
-    if (autopad && autopad != "NOTSET")
-      return rewriter.notifyMatchFailure(op, "padding must be explicit");
+    if (inputType.getShape().size() != 4) {
+      return rewriter.notifyMatchFailure(convOp, "TOSA only supports conv 2d");
+    }
 
     // Convert input [N,IC,IH,IW] -> [N,IH,IW,IC]
     Value newInput =
-        tosa::createTosaTransposedTensor(rewriter, op, input, {0, 2, 3, 1});
+        tosa::createTosaTransposedTensor(rewriter, convOp, input, {0, 2, 3, 1});
 
     // Convert weights [OC,IC,KH,KW] -> [OC,KH,KW,IC]
-    Value newWeight =
-        tosa::createTosaTransposedTensor(rewriter, op, weights, {0, 2, 3, 1});
+    Value newWeight = tosa::createTosaTransposedTensor(
+        rewriter, convOp, weights, {0, 2, 3, 1});
 
     if (bias.getType().isa<NoneType>()) {
       DenseElementsAttr newBiasAttr = DenseElementsAttr::get(
           RankedTensorType::get({weightShape[0]}, rewriter.getF32Type()),
           {0.0F});
       bias = rewriter.create<mlir::tosa::ConstOp>(
-          op->getLoc(), newBiasAttr.getType(), newBiasAttr);
-    }
-    if (!dilations) {
-      dilations = rewriter.getI64ArrayAttr({1, 1});
-    }
-    if (!strides) {
-      strides = rewriter.getI64ArrayAttr({1, 1});
-    }
-    if (!pads) {
-      pads = rewriter.getI64ArrayAttr({0, 0, 0, 0});
-    } else {
-      llvm::SmallVector<int64_t, 4> newPadVec = extractFromI64ArrayAttr(pads);
-      pads = rewriter.getI64ArrayAttr(
-          {newPadVec[0], newPadVec[2], newPadVec[1], newPadVec[3]});
+          convOp->getLoc(), newBiasAttr.getType(), newBiasAttr);
     }
 
+    ArrayAttr dilations = rewriter.getI64ArrayAttr(shapeHelper.dilations);
+    ArrayAttr strides = rewriter.getI64ArrayAttr(shapeHelper.strides);
+    llvm::SmallVector<int64_t, 4> transposedPads =
```

Comment:
Please align the naming of original and transposed variables - above you used `weight` and `newWeight`.

Reply:
I will also change this in the maxpool conversion